tests/integration/test_logcollector/test_macos/test_macos_file_status_when_no_macos.py

```diff
@@ -0,0 +1,142 @@
+configuration_ids = [f"{x['FILE_TO_MONITOR']}" for x in parameters]
+
+# Max number of characters to be displayed in the log's debug message
+sample_log_length = 100
+# Time in seconds to update the file_status.json
+file_status_update_time = 4
+
+local_internal_options = { 'logcollector.vcheck_files': file_status_update_time,
+                           'logcollector.sample_log_length': sample_log_length }
+
+# Maximum waiting time in seconds to find the logs on ossec.log
+file_monitor_timeout = 30
+
+# Time to wait for file_status.json to be updated
+wait_file_status_update_time = file_status_update_time + 2
+
+wazuh_log_monitor = None
+
+
+# Fixtures
+@pytest.fixture(scope='module', params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope='module')
+def restart_required_logcollector_daemons():
+    """Wazuh logcollector daemons handler."""
+
+    required_logcollector_daemons = ['wazuh-logcollector']
+
+    for daemon in required_logcollector_daemons:
+        control_service('stop', daemon=daemon)
+
+    truncate_file(LOG_FILE_PATH)
+    os.remove(LOGCOLLECTOR_FILE_STATUS_PATH) if os.path.exists(LOGCOLLECTOR_FILE_STATUS_PATH) else None
+
+    for daemon in required_logcollector_daemons:
+        control_service('start', daemon=daemon)
+
+    yield
+
+    for daemon in required_logcollector_daemons:
+        control_service('stop', daemon=daemon)
+
+
+@pytest.fixture(scope='module')
+def startup_cleanup():
+    """Truncate local_internals file and create dummy file to be monitored by logcollector."""
+    truncate_file(WAZUH_LOCAL_INTERNAL_OPTIONS)
+
+    with open(dummy_file, 'w') as f:
+            pass
+
+
+def callback_logcollector_started():
+    """Check for 'macos' key."""
+    return make_callback(pattern='Started', prefix=logcollector_prefix)
+
+
+def test_macos_file_status_when_no_macos(startup_cleanup,
+                                         configure_local_internal_options_module,
+                                         get_configuration,
+                                         configure_environment,
+                                         restart_required_logcollector_daemons):
+    """Checks that logcollector does not store and removes, if exists, previous "macos"-formatted localfile data in the
+    file_status.json
+
+    Given a file_status.json that contains a valid combination of "settings" and "timestamp" of "macos", when starting
+    an agent that has no "macos" localfile configured on its ossec.conf file, it should happen that, when
+    file_status.json is updated after a certain time, no "macos" status should remain stored on the status file.
+
+    Raises:
+        TimeoutError: If the callbacks, that checks the expected logs, are not satisfied in the expected time.
+    """
+
+    file_status_json = {}
```

Comment:
```suggestion

```

Reply:
this is needed for creating the macos' json block that will be stored on the file_status.json